types/constants.go

```diff
@@ -8,31 +8,43 @@ package types
 
 import (
 	"math/big"
+	"sync"
 	"time"
 
 	"github.com/NebulousLabs/Sia/build"
 )
 
+// COMPATv0.4.0 - until the first 10e3 blocks have been archived, CurrentHeight
+// is needed to keep code consistent while verifying earlier blocks.
+//
+// CurrentHeight is a global variable to help with an upcoming hardfork. When
+// the height is less than 12,000, one set of code will be run. When the height
+// is greater than 12,000, a different set of code will be run. After the
+// hardfork has passed and everyone has upgraded, a checkpoint will be created
+// and blocks below 12,000 will never be viewed again.
+var CurrentHeight BlockHeight = 0
+var CurrentHeightLock sync.Mutex
```

Comment:
is CurrentHeight ever modified by more than one goroutine? Seems like it only gets modified when a new block is added/removed, which should be strictly serial.

is there a reason why the compatibility code has to live in the types package? Wouldn't it make more sense in consensus?